streams/src/main/java/org/apache/kafka/streams/kstream/KCogroupedStream.java

```diff
@@ -0,0 +1,194 @@
+     * Thus, {@code aggregate(Initializer, Aggregator, Materialized)} can be used to compute
+     * aggregate functions like count.
+     * <p>
+     * Not all updates might get sent downstream, as an internal cache is used to deduplicate
+     * consecutive updates to the same key. The rate of propagated updates depends on your input
+     * data rate, the number of distinct keys, the number of parallel running Kafka Streams
+     * instances, and the {@link StreamsConfig configuration} parameters for {@link
+     * StreamsConfig#CACHE_MAX_BYTES_BUFFERING_CONFIG cache size}, and {@link
+     * StreamsConfig#COMMIT_INTERVAL_MS_CONFIG commit intervall}.
+     * <p>
+     * To query the local {@link KeyValueStore} it must be obtained via {@link
+     * KafkaStreams#store(String, QueryableStoreType) KafkaStreams#store(...)}:
+     * <pre>{@code
+     * KafkaStreams streams = ... // some aggregation on value type double
+     * String queryableStoreName = "storeName" // the store name should be the name of the store as defined by the Materialized instance
+     * ReadOnlyKeyValueStore<String, Long> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<String, Long>keyValueStore());
+     * String key = "some-key";
+     * Long aggForKey = localStore.get(key); // key must be local (application state is shared over all running Kafka Streams instances)
+     * }</pre>
+     * For non-local keys, a custom RPC mechanism must be implemented using {@link
+     * KafkaStreams#allMetadata()} to query the value of the key on a parallel running instance of
+     * your Kafka Streams application.
+     *
+     * <p>
+     * For failure and recovery the store will be backed by an internal changelog topic that will be
+     * created in Kafka. Therefore, the store name defined by the Materialized instance must be a
+     * valid Kafka topic name and cannot contain characters other than ASCII alphanumerics, '.', '_'
+     * and '-'. The changelog topic will be named "${applicationId}-${storeName}-changelog", where
+     * "applicationId" is user-specified in {@link StreamsConfig} via parameter {@link
+     * StreamsConfig#APPLICATION_ID_CONFIG APPLICATION_ID_CONFIG}, "storeName" is the provide store
+     * name defined in {@code Materialized}, and "-changelog" is a fixed suffix.
+     * <p>
+     * You can retrieve all generated internal topic names via {@link Topology#describe()}.
+     *
+     * @param initializer  an {@link Initializer} that computes an initial intermediate aggregation
+     *                     result
+     * @param storeSupplier an {@link StoreSupplier} that set up the state store.
+     * @return a {@link KTable} that contains "update" records with unmodified keys, and values that
+     * represent the latest (rolling) aggregate for each key
+     */
+    KTable<K, V> aggregate(final Initializer<V> initializer,
+                           final StoreSupplier<KeyValueStore> storeSupplier);
```

Comment:
We don't expose `StoreSupplier` in the DSL any longer -- I think this overload can be removed.